packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -78,21 +132,21 @@ export abstract class VirtualGatewayListener {
    * Returns an HTTP Listener for a VirtualGateway
    */
   public static http(options: HttpGatewayListenerOptions = {}): VirtualGatewayListener {
-    return new VirtualGatewayListenerImpl(Protocol.HTTP, options.healthCheck, options.port, options.tlsCertificate);
+    return new VirtualGatewayListenerImpl(Protocol.HTTP, options.healthCheck, options.port, options.tlsCertificate, options.connectionPool);
   }
 
   /**
    * Returns an HTTP2 Listener for a VirtualGateway
    */
-  public static http2(options: HttpGatewayListenerOptions = {}): VirtualGatewayListener {
-    return new VirtualGatewayListenerImpl(Protocol.HTTP2, options.healthCheck, options.port, options.tlsCertificate);
+  public static http2(options: Http2GatewayListenerOptions = {}): VirtualGatewayListener {
```

Comment:
This change needs to go into the "BREAKING CHANGES" section of the doc (it's not breaking for JS/TS, but it is for other languages like Java).